R/convert-gene-ids.R

```diff
@@ -0,0 +1,125 @@
+#' Convert Ensembl gene ids to gene symbols based on an ScPCA SingleCellExperiment object
+#'
+#' The SingleCellExperiment objects produced as part of ScPCA are indexed by
+#' Ensembl gene ids, as those are more stable than gene symbols. However,
+#' for many applications gene symbols are useful. This function provides a
+#' simple and consistent conversion of Ensembl gene ids to gene symbols based on
+#' the `gene_symbol` column that is present in the row data of ScPCA
+#' SingleCellExperiment objects.
+#'
+#' For this function, the SingleCellExperiment object must contain a `gene_ids`
+#' column containing Ensembl gene ids and a `gene_symbol` column containing gene
+#' symbols. If any gene ids are not found or if the gene symbol is not defined,
+#' the input gene id is returned, unless the `leave_na` is set to `TRUE`.
+#'
+#'
+#' @param ensembl_ids A character vector of Ensembl gene ids to translate to
+#'  gene symbols.
+#' @param sce A SingleCellExperiment object containing gene ids and gene symbols
+#'  to use for translation.
+#' @param leave_na logical indicating whether to leave NA values in the output.
+#'  Default is `FALSE`
+#'
+#' @return A vector of gene symbols corresponding to the input Ensembl ids.
+#' @export
+#'
+#' @import SingleCellExperiment
+#'
+#' @examples
+#' \dontrun{
+#' # convert a set of Ensembl ids to gene symbols
+#' # using a SingleCellExperiment reference
+#' ensembl_ids <- c("ENSG00000141510", "ENSG00000134323")
+#' gene_symbols <- ensembl_to_symbol(ensembl_ids, sce)
+#' gene_symbols
+#' ### [1] "TP53" "MYCN"
+#' }
+ensembl_to_symbol <- function(ensembl_ids, sce, leave_na = FALSE) {
```

Comment:
Given that there's already a bit of a level of abstraction here, is there a reason to not pass in a gene_symbol vector here rather than the full sce? I may be answering my own question - I suspect to confirm that the source of the gene symbols is our SCE object as we have mapped symbols and ids already?

Reply:
Yes, I don't want people to have to extract anything.